### ISSUE TITLE
AttributeError: 'NoneType' object has no attribute 'peer'

### ISSUE THREAD

**Issue by mill1000** [mill1000/midea-msmart#155]
I've seen this exception being thrown in a handful of logs.

https://github.com/mill1000/midea-ac-py/issues/187#issuecomment-2253206067
https://github.com/mill1000/midea-ac-py/issues/188

```
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 312, in _async_refresh
    self.data = await self._async_update_data()
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/midea_ac/coordinator.py", line 39, in _async_update_data
    await self._device.refresh()
  File "/usr/local/lib/python3.12/site-packages/msmart/device/AC/device.py", line 389, in refresh
    for response in await self._send_command_get_responses(cmd):
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/msmart/device/AC/device.py", line 278, in _send_command_get_responses
    responses = await super()._send_command(command)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/msmart/base_device.py", line 38, in _send_command
    responses = await self._lan.send(data)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/msmart/lan.py", line 617, in send
    self._protocol.peer)
    ^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'peer'
```

I can't wrap my ahead around what's causing it. Only a call to `disconnect` should make the protocol be None and `disconnect` is called within the exception handler which should throw and break it out of the loop

**Comment by mill1000** [comment]
Another use with lots of assertion failures
https://github.com/mill1000/midea-ac-py/issues/198#issuecomment-2264839750

Always seems like enabling debug logging causes them to go away. Race condition?

**Comment by mill1000** [comment]
Also in this log
https://github.com/mill1000/midea-ac-py/issues/212#issue-2459274710

**Comment by mill1000** [comment]
Possible explanation here: https://github.com/python/cpython/issues/114496

**Comment by mill1000** [comment]
Hopefully fixed in #167
